core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -173,6 +175,31 @@ public static void merge(Map<?, ?> properties) {
     checkUserFileBufferBytes();
   }
 
+  /**
+   * Iterates a set of site properties and discards those that user should not use.
+   *
+   * @param siteProperties the set of site properties to check
+     */
+  public static void discardIgnoredSiteProperties(Map<?, ?> siteProperties) {
+    if (siteProperties != null) {
```

Comment:
when would siteProperties be null here?

Reply:
It will be `null` if `Configuration` fails to pick up a site properties file either because the path to search for the file is missing, or there is no such file. Since we perform a null check before calling this method, I don't have a strong opinion about keeping or discarding this check inside the method.

Reply:
ok, lets remove the null check if it's not necessary. In general `null` is not a valid argument unless the function explicitly documents how it will handle null